### PR TITLE
Grace period as a Saga

### PR DESCRIPTION
**Pull request by dvdstelt**
solves #7 

This implements the grace period using a saga.

### CODE REVIEW

**Comment by dvdstelt** [comment]
So far I did 2 commits, but I stopped and looked again really hard at the code to see how we can do this properly. That's why I also created the [sequence diagram](https://www.websequencediagrams.com/). Code for it is in issue #26.

I will continue this Monday if @kbaley doesn't work on it today.

File: src/Services/Ordering/Ordering.API/Application/Sagas/GracePeriod.cs [hunk 1]
```diff
@@ -41,7 +42,6 @@ protected override void ConfigureMapping(IMessagePropertyMapper mapper)
             // - Verify if there is stock available
             var @event = new OrderStatusChangedToAwaitingValidationIntegrationEvent(message.OrderId, message.OrderedItems);
             await context.Publish(@event);
-            await RequestTimeout<GracePeriodExpired>(context, TimeSpan.FromMinutes(settings.GracePeriodTime));
```
Comment (dvdstelt) [inline src/Services/Ordering/Ordering.API/Application/Sagas/GracePeriod.cs]: @kbaley Why did you remove this?

File: src/Services/Ordering/Ordering.API/Application/Commands/CancelOrderCommandHandler.cs [hunk 2]
```diff
@@ -24,22 +27,24 @@ protected override bool CreateResultForDuplicateRequest()
     public class CancelOrderCommandHandler : IAsyncRequestHandler<CancelOrderCommand, bool>
     {
         private readonly IOrderRepository _orderRepository;
+        private readonly IEndpointInstance _endpoint;
```
Comment (mauroservienti) [inline src/Services/Ordering/Ordering.API/Application/Commands/CancelOrderCommandHandler.cs]: Should this be `IMessageSession` instead of `IEndpointInstance`?

**Comment by mauroservienti** [comment]
Feel free to merge whenever you want @kbaley

File: src/Services/Ordering/Ordering.API/Application/Commands/CancelOrderCommandHandler.cs [hunk 3]
```diff
@@ -24,22 +27,24 @@ protected override bool CreateResultForDuplicateRequest()
     public class CancelOrderCommandHandler : IAsyncRequestHandler<CancelOrderCommand, bool>
     {
         private readonly IOrderRepository _orderRepository;
+        private readonly IMessageSession _endpoint;
+
 
-        public CancelOrderCommandHandler(IOrderRepository orderRepository)
+        public CancelOrderCommandHandler(IOrderRepository orderRepository, IEndpointInstance endpoint)
```
Comment (mauroservienti) [inline src/Services/Ordering/Ordering.API/Application/Commands/CancelOrderCommandHandler.cs]: This one has to be `IMessageSession` too
